app/controllers/admin/bookmarks_controller.rb

```diff
@@ -0,0 +1,76 @@
+class Admin::BookmarksController < AdminController
+  before_action :set_count
+  before_action :set_bookmark, only: [:show, :edit, :update, :destroy]
+
+  # GET /bookmarks
+  def index
+    @bookmarks = Bookmark.all.order(created_at: :desc).page params[:page]
+  end
+
+  # GET /bookmarks/1
+  def show
+    respond_to do |format|
+      format.html { render :show }
+    end
+  end
+
+  # GET /bookmarks/new
+  # def new
+  #   @bookmark = Bookmark.new
+  # end
+
+  # POST /bookmarks
+  # def create
+  #   @bookmark = Bookmark.new bookmark_params
+
+  #   respond_to do |format|
+  #     if @bookmark.save
+  #       format.html { redirect_to [:admin, @bookmark], notice: "Bookmark was successfully created." }
+  #     else
+  #       format.html { render :new }
+  #     end
+  #   end
+  # end
+
+  # GET /bookmarks/1/edit
+  def edit
+  end
+
+  # PATCH/PUT /bookmarks/1
+  def update
+    respond_to do |format|
+      if @bookmark.update(bookmark_params)
+        format.html { redirect_to [:admin, @bookmark], notice: "Bookmark was successfully updated." }
+      else
+        format.html { render :edit }
+      end
+    end
+  end
+
+  # DELETE /bookmarks/1
+  # def destroy
+  #   respond_to do |format|
+  #     if @bookmark.destroy
+  #       format.html { redirect_to admin_bookmarks_url, notice: "Bookmark was successfully deleted." }
+  #     else
+  #       format.html { render :new }
+  #     end
+  #   end
+  # end
+
+  private
+
+  def set_bookmark
+    @bookmark = Bookmark.find params[:id]
+  end
+
+  def set_count
+    @count = Bookmark.count
```

Comment:
What's going on here? Counting all the bookmarks in the system? Or is there an implicit scope of current user or something?

This will become slow if you get over a million or bajillion bookmarks.

Reply:
when it gets slow I'll make a job to update a cached count or use one of the fancy postgres quick count things :P It shows up in a little badge on the menu
![transientbug admin bookmarks 2018-01-17 11-00-37](https://user-images.githubusercontent.com/94542/35058598-b1ab971a-fb75-11e7-91b2-07f1520a3f6a.png)

Reply:
> postgres quick count things

go on

Reply:
https://wiki.postgresql.org/wiki/Count_estimate estimating can be quick. And there is probably a point at which I'll remove the count badge cause it gets beyond ridiculous